docs/advanced-features/sitemaps.md

```diff
@@ -0,0 +1,198 @@
+---
+description: Learn how to add a sitemap to your Next.js project, along with a robots.txt file for optimal SEO.
+---
+
+# Sitemaps
+
+<details open>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/with-sitemap">with-sitemap</a></li>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/with-next-sitemap">with-next-sitemap</a></li>
+  </ul>
+</details>
+
+To improve your Search Engine Optimization (SEO), you might want to add a [sitemap](https://developers.google.com/search/docs/advanced/sitemaps/build-sitemap) or [`robots.txt`](https://developers.google.com/search/docs/advanced/robots/intro) file to your Next.js website.
+
+A **sitemap** defines the relationship between pages of your site. Search engines utilize
+this file to more accurately index your site. You can also provide additional information
+such as last updated time, how frequently the page changes, and more.
+
+A **robots.txt** file tells search engines which pages or files the crawler can or can't request from your site.
+
+## Static Sitemaps
+
+If your website does not update frequently, you can manually create a static sitemap.
+This is a basic `.xml` file defining the content of your site. Here's a simple example:
+
+```xml
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+  <url>
+      <loc>https://nextjs.org</loc>
+  </url>
+  <url>
+      <loc>https://nextjs.org/docs</loc>
+  </url>
+</urlset>
+```
+
+## Dynamic Sitemaps
+
+If your site frequently changes, you should dynamically create a sitemap. First, let's explore an example where your content is based on [file-system routing](/docs/routing/introduction.md) (e.g., contained inside the `/pages` directory).
+
+This example is using [globby](https://github.com/sindresorhus/globby) to fetch our list of routes.
+
+```bash
+npm install globby --save-dev
+```
+
+Next, we can create a Node script at `scripts/generate-sitemap.mjs`.
+This file will dynamically build a sitemap based on your `/pages` directory.
+
+```js
+import { writeFileSync } from 'fs'
+import { globby } from 'globby'
+
+async function generate() {
+  const pages = await globby([
+    'pages/*.js', // Include all pages
+    '!pages/_*.js', // Exclude _app or _document
+    '!pages/api', // Exclude API Routes
+    '!pages/404.js', // Exclude 404
+  ])
+
+  const sitemap = `
```

Comment:
in case you're open to using a third-party library for constructing the xml string - the `unified` ecosystem has [`xast-util-sitemap`](https://github.com/syntax-tree/xast-util-sitemap)